File: JAVA/NimArray.java
import java.util.*;

public class NimArray {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Masukkan NIM: ");
        String nim = sc.next();

        ArrayList<Integer> digits = new ArrayList<>();

        // Memecah NIM
        for (char c : nim.toCharArray()) {
            digits.add(c - '0');
        }

        // Perhitungan
        int total = 0;
        for (int d : digits) total += d;

        int maksimum = Collections.max(digits);
        int minimum = Collections.min(digits);
        double rata = (double) total / digits.size();

        ArrayList<Integer> reverse = new ArrayList<>(digits);
        Collections.reverse(reverse);

        System.out.println("Array digit: " + digits);
        System.out.println("Total digit = " + total);
        System.out.println("Digit maksimum = " + maksimum);
        System.out.println("Digit minimum = " + minimum);
        System.out.println("Rata-rata = " + rata);
        System.out.println("Reverse = " + reverse);
    }
}

File: PYTHON/nim_array.py
nim = input("Masukkan NIM: ")

# Memecah NIM
digits = [int(c) for c in nim]k

# Perhitungan
total = sum(digits)
maksimum = max(digits)
minimum = min(digits)
rata = total / len(digits)
reverse = digits[::-1]

print("Array digit:", digits)
print("Total digit =", total)
print("Digit maksimum =", maksimum)
print("Digit minimum =", minimum)
print("Rata-rata =", rata)
print("Reverse =", reverse)